### PR TITLE
Added a new API version to the spec which includes APNS HTTP 2 parameter for APNSCredential

### PR DESCRIPTION
**Pull request by amolr**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by balajikris** [comment]
As I understand it, this PR moves the changes added in PR https://github.com/Azure/azure-rest-api-specs/pull/1145 into a new spec with a bumped up version number.

@amolr - please correct me if i'm wrong.

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 1]
```diff
@@ -0,0 +1,1956 @@
...
+    "description": "Azure NotificationHub client",
+    "version": "2017-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Could you please check and confirm if your service really supports text/json????

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 2]
```diff
@@ -0,0 +1,1956 @@
...
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.NotificationHubs/checkNamespaceAvailability": {
+      "post": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_CheckAvailability",
+        "description": "Checks the availability of the given service namespace across all Azure subscriptions. This is useful because the domain name is created based on the service namespace name.",
+        "externalDocs": {
+          "url": "http://msdn.microsoft.com/en-us/library/windowsazure/jj870968.aspx"
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: In the specified url the x-ms-version header is marked as 2012-03-01. There is no such mention in this file.

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 3]
```diff
@@ -0,0 +1,1956 @@
...
+      },
+      "get": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_GetAuthorizationRule",
+        "description": "Gets an authorization rule for a namespace by name.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Can these 3 parameters (resourceGroupName, namespaceName, authorizationRuleName) be defined globally and referenced here?

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 4]
```diff
@@ -0,0 +1,1956 @@
...
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.NotificationHubs/namespaces": {
+      "get": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_ListAll",
+        "description": "Lists all the available namespaces within the subscription irrespective of the resourceGroups.",
+        "externalDocs": {
+          "url": "http://msdn.microsoft.com/en-us/library/azure/hh780759.aspx"
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: This api is to list namespaces. The link is for list queues. Please fix it. Also check all other links also,

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 5]
```diff
@@ -0,0 +1,1956 @@
...
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": " Gets all the Namespaces within the subscription irrespective of the ResourceGroup",
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: nit: space in the beginnng of the description

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 6]
```diff
@@ -0,0 +1,1956 @@
...
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Get the Namespace AuthorizationRule",
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: It is actually "Get the List of AuthorizationRules of a Namespace"

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 7]
```diff
@@ -0,0 +1,1956 @@
...
+            "description": "The connection string of the NotificationHub for the specified authorizationRule."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: The previous 2 operations also return a list and are pageable. It seems this one also returns a list and is NOT pageable. Is this intentional?

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 8]
```diff
@@ -0,0 +1,1956 @@
...
+          "200": {
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Same as abve.

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json [hunk 9]
```diff
@@ -0,0 +1,1956 @@
...
+            "schema": {
+              "$ref": "#/definitions/PnsCredentialsResource"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "CheckAvailabilityParameters": {
```
Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: The best way to model this is to make it allOf on Resource.

Comment (sarangan12) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Also, if you do the above change, this model would be exactly equivalent to the next model CheckAvailabilityResult. Then why do you need 2 model definitions?

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 2 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: The URL is for the old RDFE API. There is no documentation for the new API yet

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 1 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Need to check all the service calls for this one. Will take a separate work item

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 2 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Removed the externalDocs parameter

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 3 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Will cause a breaking change in the SDK. Will take as a separate work item

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 4 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Removed the external docs parameter

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 5 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Done

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 6 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Done

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 7 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: The ListKeys are not going to that high

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 8 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: Same here

File: arm-notificationhubs/2017-04-01/swagger/notificationhubs.json (hunk 9 above)
Comment (amolr) [inline arm-notificationhubs/2017-04-01/swagger/notificationhubs.json]: We have a separate work item to track this. Already changed the service code. We will make the necessary changes soon

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**